src/main/java/io/vertx/core/shareddata/impl/LocalAsyncMapImpl.java

```diff
@@ -93,9 +93,7 @@ private void removeIfExpired(K k) {
     long timerId = vertx.setTimer(ttl, l -> removeIfExpired(k));
     Holder<V> existing = map.putIfAbsent(k, new Holder<>(v, timerId, ttl, timestamp));
     if (existing != null) {
-      if (existing.expires()) {
-        vertx.cancelTimer(timerId);
-      }
+      vertx.cancelTimer(timerId);
```

Comment:
Nice catch! Can you please:

- add a new test case for this
- file a separate issue to keep track of the bug (that we will close by merging this PR)

We use issues to generate the release notes so a dedicated issue will make it clear that a bug was fixed.

Reply:
i tried to test it now... old and new and it actually never happens because even if the new timer tries to delete the old record it cannot be deleted if not put with ttl (not expiring old)

so it's more a cosmetic cleanup

Reply:
@tsegismont added the test case but should i still create an issue for it?

Reply:
@zyclonite no, it's fine